firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigSharedPrefsClient.java

```diff
@@ -249,6 +257,46 @@ void setBackoffMetadata(int numFailedFetches, Date backoffEndTime) {
     }
   }
 
+  public void setCustomSignals(Map<String, Object> newCustomSignals) {
+    synchronized (customSignalsLock) {
+      // Retrieve existing custom signals
+      Map<String, Object> existingCustomSignals = getCustomSignals();
+
+      for (Map.Entry<String, Object> entry : newCustomSignals.entrySet()) {
+        String key = entry.getKey();
+        Object value = entry.getValue();
```

Comment:
Can we check the Object type here? If it's not long or string we shouldn't save it according to the comment on public api.

Reply:
Thanks, I have added a check for this.